2-WebApp-graph-user/2-5-HybridFlow/wwwroot/js/HybridSPA.js

```diff
@@ -0,0 +1,101 @@
+﻿function getTokenFromCache() {
+    const [account] = msalInstance.getAllAccounts();
+    const scopes = ["user.read"];
+    return msalInstance.acquireTokenSilent({
+        account,
+        scopes
+    });
+};
+
+const profileDiv = document.getElementById("profile-div");
+
+function loadUserData() {
+    getTokenFromCache()
+        .then(response => {
+            callMSGraph('https://graph.microsoft.com/v1.0/me/messages', response.accessToken)
```

Comment:
nit : can this be added to config?

Reply:
The short answer is _yes_.

The long answer is that I'd like to come back and refactor this. If I want to port values from the configuration directly into the application it'll need to be pre-rendered.

I'll scope this out.